core/block/editor/basic/history.go

```diff
@@ -2,14 +2,15 @@ package basic
 
 import (
 	"github.com/anyproto/anytype-heart/core/block/editor/smartblock"
+	"github.com/anyproto/anytype-heart/core/block/undo"
 	"github.com/anyproto/anytype-heart/core/session"
 	"github.com/anyproto/anytype-heart/pb"
 	"github.com/anyproto/anytype-heart/util/pbtypes"
 )
 
 type IHistory interface {
-	Undo(*session.Context) (counters pb.RpcObjectUndoRedoCounter, err error)
-	Redo(*session.Context) (counters pb.RpcObjectUndoRedoCounter, err error)
+	Undo(*session.Context) (counters pb.RpcObjectUndoRedoCounter, carriageInfo undo.CarriageInfo, err error)
```

Comment:
maybe wrap return result into structure {counter, carriageInfo}?

Reply:
Created **HistoryInfo** wrapper to pass info from undo service up to RpcHandler